src/vs/workbench/contrib/workspace/browser/workspaceTrustEditor.ts

```diff
@@ -783,7 +814,7 @@ export class WorkspaceTrustEditor extends EditorPane {
 	private addTrustButtonToElement(parent: HTMLElement): void {
 		const trustUris = async (uris?: URI[]) => {
 			if (!uris) {
-				await this.workspaceTrustManagementService.setWorkspaceTrust(true);
+				await this.workspaceTrustManagementService.setWorkspaceTrust(true, acceptsNonWorkspaceFiles(this.editorService, this.workspaceService));
```

Comment:
this part seems to complicate our API and I wonder if it should. when the user trusts a workspace, they may think about all the editors they have open, but they may not consider that they have already introduced untrusted files so I'm not sure we should be smart here and instead keep the API simple and prompt if additional files are introduced.

Reply:
Agreed. Removed it.